### PR TITLE
Added Atuh widget

### PR DESCRIPTION
**Pull request by jeetbhalani15**
## Added Authentication
- Added Login , Signup, & Logout pages
- Added Context & API Calls for auth
- With added Form Validations

Ignore Package.json Files & Colors are hard coded change it soon after dark mode

image

### CODE REVIEW

File: Gravity-tube/src/Pages/AuthWidget/Login/Login.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    email: "",
+    password: "",
+  };
+
+  const dummyData = {
+    email: "guest@gmail.com",
+    password: "Abc@1234",
+  };
+
+  const formReducer = (state, { type, payload }) => {
```
Comment (Rohitprasad83) [inline Gravity-tube/src/Pages/AuthWidget/Login/Login.js:26]: put this reducer function to a different file.

File: Gravity-tube/src/Pages/AuthWidget/Login/Login.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+      default:
+        return state;
+    }
+  };
+
+  const [formState, formDispatch] = useReducer(formReducer, initialState);
+
+  const LoginHandler = async (e) => {
+    e.preventDefault();
+    console.log(formState.email)
```
Comment (Rohitprasad83) [inline Gravity-tube/src/Pages/AuthWidget/Login/Login.js:43]: remove console.logs()

File: Gravity-tube/src/Pages/AuthWidget/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+        theme: "dark",
+        });
+    }
+  };
+  return (
+      <div className="background">
+        <div className="login-box">
+          <div className="login-content">
+            {/* <div className="auth-logo">
+              <div className="logo-mg">
```
Comment (Rohitprasad83) [inline Gravity-tube/src/Pages/AuthWidget/Login/Login.js:85]: remove commented code

File: Gravity-tube/src/Pages/AuthWidget/SignUp/SignUp.js [hunk 4]
```diff
@@ -0,0 +1,389 @@
...
+          closeOnClick: true,
+          pauseOnHover: true,
+          draggable: true,
+          progress: undefined,
+          theme: "dark",
+          });
+      }
+    }
+  };
+
```
Comment (Rohitprasad83) [inline Gravity-tube/src/Pages/AuthWidget/SignUp/SignUp.js:175]: put all the required utility functions like formValidation to a separate file

File: Gravity-tube/src/Pages/AuthWidget/SignUp/SignUp.js [hunk 5]
```diff
@@ -0,0 +1,389 @@
...
+  return (
+    <div className="html">
+      <div className="background">
+        <div className="login-box">
+          <div className="login-content">
+            {/* <div className="auth-logo">
+              <div className="logo-mg">
+                <img className="logo" src={logo} alt="logo" />
```
Comment (Rohitprasad83) [inline Gravity-tube/src/Pages/AuthWidget/SignUp/SignUp.js:183]: Commented code

**Review by Rohitprasad83 — APPROVED** [review]
Code can be refactored, separate files can be used for utility functions and reducer functions.
- code is easily readable
- good naming convention is used

File: Gravity-tube/src/Pages/AuthWidget/Login/Login.js (hunk 2 above)
Comment (jeetbhalani15) [reply Gravity-tube/src/Pages/AuthWidget/Login/Login.js:43]: ok sure

File: Gravity-tube/src/Pages/AuthWidget/Login/Login.js (hunk 1 above)
Comment (jeetbhalani15) [reply Gravity-tube/src/Pages/AuthWidget/Login/Login.js:26]: yeah sure

File: Gravity-tube/src/Pages/AuthWidget/Login/Login.js (hunk 3 above)
Comment (jeetbhalani15) [reply Gravity-tube/src/Pages/AuthWidget/Login/Login.js:85]: ok

File: Gravity-tube/src/Pages/AuthWidget/SignUp/SignUp.js (hunk 4 above)
Comment (vivek-guda) [reply Gravity-tube/src/Pages/AuthWidget/SignUp/SignUp.js:175]: Move reducer functions to separate files

File: Gravity-tube/src/Pages/AuthWidget/SignUp/SignUp.js [hunk 6]
```diff
@@ -0,0 +1,389 @@
...
+              </div>
+              <div className="bg-color">
+                ZERO
+                <div>GRAVITY</div>
+                <small className="small-txt">Store</small>
+              </div>
+            </div> */}
```
Comment (vivek-guda) [inline Gravity-tube/src/Pages/AuthWidget/SignUp/SignUp.js:190]: Try clearing the commented code

**Comment by vivek-guda** [comment]
UI looks good

**Review by Jay-Gadhiya — APPROVED** [review]
- you can change the suggested changes
- code looks clean
- signup looks good

**Pull request merged**